fieldtracing/fieldtracing.cpp

```diff
@@ -996,114 +1010,213 @@ namespace FieldTracing {
+               }
+               for(int n=0; n<smallSizeFW; n++) {
+                  reducedCellFWTracingStepSize[indicesToReduceFW[n]] = smallReducedCellFWTracingStepSize[n];
+                  reducedCellFWRunningDistance[indicesToReduceFW[n]] = smallReducedCellFWRunningDistance[n];
+                  reducedCellNeedsContinuedFWTracing[indicesToReduceFW[n]] = smallReducedCellNeedsContinuedFWTracing[n];
+                  reducedCellFWConnection[indicesToReduceFW[n]] = smallReducedCellFWConnection[n];
+                  sumCellFWTracingCoordinates[indicesToReduceFW[n]] = smallSumCellFWTracingCoordinates[n];
+               }
+               for(int n=0; n<smallSizeBW; n++) {
+                  reducedCellBWTracingStepSize[indicesToReduceBW[n]] = smallReducedCellBWTracingStepSize[n];
+                  reducedCellBWRunningDistance[indicesToReduceBW[n]] = smallReducedCellBWRunningDistance[n];
+                  reducedCellNeedsContinuedBWTracing[indicesToReduceBW[n]] = smallReducedCellNeedsContinuedBWTracing[n];
+                  reducedCellBWConnection[indicesToReduceBW[n]] = smallReducedCellBWConnection[n];
+                  sumCellBWTracingCoordinates[indicesToReduceBW[n]] = smallSumCellBWTracingCoordinates[n];
                }
-               anyCellNeedsTracing = false;
             }
             #pragma omp barrier
             phiprof::stop("MPI-loop");
             #pragma omp single
             {
-               cellsToDo = 0;
+               cellsToDoFullBox = 0;
+               cellsToDoFluxRopes = 0;
+               cellFWTracingStepSize = reducedCellFWTracingStepSize;
+               cellBWTracingStepSize = reducedCellBWTracingStepSize;
+               cellFWRunningDistance = reducedCellFWRunningDistance;
+               cellBWRunningDistance = reducedCellBWRunningDistance;
+               cellNeedsContinuedFWTracing = reducedCellNeedsContinuedFWTracing;
+               cellNeedsContinuedBWTracing = reducedCellNeedsContinuedBWTracing;
+               cellFWConnection = reducedCellFWConnection;
+               cellBWConnection = reducedCellBWConnection;
             }
-            #pragma omp for schedule(dynamic) reduction(||:anyCellNeedsTracing) reduction(+:cellsToDo)
+            #pragma omp for schedule(dynamic) reduction(+:cellsToDoFullBox) reduction(+:cellsToDoFluxRopes)
             for(int n=0; n<globalDccrgSize; n++) {
-               if(reducedCellNeedsContinuedFWTracing[n] > 0) {
-                  anyCellNeedsTracing=true;
-                  cellNeedsContinuedFWTracing[n] = 1;
-                  cellsToDo++;
-                  
+               if(cellNeedsContinuedFWTracing[n] > 0) {
+                  cellsToDoFullBox++;
                   // Update that nodes' tracing coordinates
                   cellFWTracingCoordinates[n][0] = sumCellFWTracingCoordinates[n][0] / reducedCellNeedsContinuedFWTracing[n];
```

Comment:
What's going on here? Isn't that a division by 1 always (when inside this if)?

Reply:
There was a version of the code where that could be more than 1. I ran a quick test that confirmed this isn't the case anymore.